src/Swift.Runtime/src/Library/UnsafeBufferPointer.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Swift.Runtime
+{
+    // <summary>
+    // Represents Swift UnsafeRawBufferPointer in C#.
+    // </summary>
+    public unsafe readonly struct UnsafeRawBufferPointer
+    {
+        private readonly void* _Position;
+        private readonly void* _End;
+        public UnsafeRawBufferPointer(void* start, nint count)
+        {
+            _Position = start;
+            _End = (byte*)start + count;
+        }
+
+        public void* BaseAddress => _Position;
+        public nint Count => (nint)((byte*)_End - (byte*)_Position);
+    }
+
+
+    // <summary>
+    // Represents Swift UnsafeMutableRawBufferPointer in C#.
+    // </summary>
+    public unsafe readonly struct UnsafeMutableRawBufferPointer
+    {
+        private readonly void* _Position;
+        private readonly void* _End;
+        public UnsafeMutableRawBufferPointer(void* start, nint count)
+        {
+            _Position = start;
+            _End = (byte*)start + count;
+        }
+
+        public void* BaseAddress => _Position;
+        public nint Count => (nint)((byte*)_End - (byte*)_Position);
+    }
+
+    // <summary>
+    // Represents Swift UnsafeBufferPointer in C#.
+    // </summary>
+    public unsafe readonly struct UnsafeBufferPointer
+    {
+        private readonly void* _Position;
+        public readonly nint Count;
```

Comment:
Swift does not seem to have fields as first-class concept. Their docs only talk about properties: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/properties/

I am wondering whether it is mapping stored properties to fields in C# is conceptually correct.

Is it breaking change in Swift to change a stored property to a computed property in non-frozen struct? I would expect that it is not a breaking change - can you confirm that?

If computed properties and stored properties are interchangeable in Swift, we should map both to properties in C#. Ie make this `public nint Count { get; };` or `private nint _count; public nint Count => _count;`.

Reply:
Good catch. This is not a breaking change in Swift. I confirmed it by consuming a library with struct that evolved from
```swift
public struct MyType {
    public var myProperty: Int = 0
    public init() {}
}
```
to
```swift
public struct MyType {
    private var _myProperty: Int = 0
    public var myProperty: Int {
        get { _myProperty }
        set { _myProperty = newValue }
    }

    public init() {}
}
```

Reply:
Buffer pointers are updated with `private readonly nint _count; public nint Count => _count;`.

Reply:
IIRC, if a property is stored, you still get getters and setter generated by the compiler. Since BTfS doesn't deal with struct lowering, it maps properties (stored or otherwise) as C# properties.